File: mt9x9.swift
/*
9x9 multiplication table in Swift
CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2018.
*/
// swiftc mt9x9.swift && ./mt9x9 || swift mt9x9.swift

import Foundation
for i in stride(from:1, to:9, by:3) {
    for j in 1...9 {
        for k in [i, i+1, i+2] {
            print(String(format:"%dx%d=%2d", k, j, k*j), terminator:"\t")
        }
        print()
    }
    print()
}

File: mt9x9.erl
% 9x9 multiplication table in Erlang
% CC0, Wei-Lun Chao <bluebat@member.fsf.org>, 2021.
% erlc mt9x9.erl && erl -noshell -s mt9x9 start -s init stop

-module(mt9x9).
-export([start/0]).
start() ->
    [block(I) || I <- lists:seq(1, 9, 3)].
block(I) ->
    [line(I, J) || J <- lists:seq(1, 9)],
    io:fwrite("~n").
line(I, J) ->
    [io:format("~Bx~B=~2B\t", [K, J, K*J]) || K <- lists:seq(I, I+2)],
    io:nl().
